examples/chip-tool/main.cpp

```diff
@@ -33,11 +33,11 @@ int main(int argc, char * argv[])
 {
     ExampleCredentialIssuerCommands credIssuerCommands;
     Commands commands;
-    registerCommandsDiscover(commands);
+    registerCommandsDiscover(commands, &credIssuerCommands);
     registerCommandsPayload(commands);
     registerCommandsPairing(commands, &credIssuerCommands);
     registerCommandsTests(commands);
-    registerClusters(commands);
+    registerClusters(commands, &credIssuerCommands);
```

Comment:
Sorry can you explain me why do we creates an `ExampleCredentialIssuerCommands` here and one into `CHIPCommand.h` that is overridden by the one passed into the constructor ?

I would really have preferred that this stuff does not lives into `main.cpp` unless there is a good reason for it. But I assume that I missed the point ?

Reply:
It will be easier for a vendor to create a new main.cpp with the desired instance for CredentialIssuerCommands rather than overriding all CHIPCommand declarations in the source code. This way the vendor only has to link the static chip-tool lib with their own main.cpp.

Cleaned up and removed CredentialIssuerCommands from CHIPCommand class. Please check the updated changes in this PR.

Reply:
@vivien-apple  Please review the latest changes

Reply:
Vivien reviewed the latest commit and gave a +1